common/client/models.go

```diff
@@ -74,3 +75,41 @@ func (n NodeTier) String() string {
 		return fmt.Sprintf("NodeTier(%d)", n)
 	}
 }
+
+type syncIssue int
+
+func (s syncIssue) String() string {
+	if s == 0 {
+		return "synced"
+	}
+	var result bytes.Buffer
+	for i := syncIssueNotInSyncWithPool; i < syncIssueLen; i = i << 1 {
+		if i&s == 0 {
+			continue
+		}
+		result.WriteString(i.string())
+		result.WriteString(",")
+	}
+	result.Truncate(result.Len() - 1)
+	return result.String()
+}
+
+func (s syncIssue) string() string {
+	switch s {
+	case syncIssueNotInSyncWithPool:
+		return "NotInSyncWithRPCPool"
+	case syncIssueNoNewHead:
+		return "NoNewHead"
+	case syncIssueNoNewFinalizedHead:
+		return "NoNewFinalizedHead"
+	default:
+		return fmt.Sprintf("syncIssue(%d)", s)
+	}
+}
+
+const (
```

Comment:
nit: can you add these at the top of `syncIssue`'s method so they're more visible?